src/components/SocialMediaLinks/SocialMediaLinks.tsx

```diff
@@ -0,0 +1,195 @@
+export type ColorType = typeof colorTypeArray[number];
+
+export const sizeTypeArray = ["small", "medium", "large"] as const;
+export type SizeType = typeof sizeTypeArray[number];
+
+export const socialMediaLinkTypeArray = [
+  "blogs",
+  "facebook",
+  "instagram",
+  "pinterest",
+  "soundcloud",
+  "tiktok",
+  "tumbler",
+  "twitter",
+  "youtube",
+] as const;
+export type SocialMediaLinkType = typeof socialMediaLinkTypeArray[number];
+
+export interface SocialMediaLinkDataProps {
+  /** Optional override for default platform name */
+  labelText?: string;
+  /** Required. Must be one of socialMediaLinkTypeArray */
+  type: SocialMediaLinkType;
+  /** Optional override for default social media url */
+  url?: string;
+}
+
+export interface SocialMediaLinksProps {
+  /** Optional border: straight, circular or none */
+  borders?: BorderType;
+  /** Optional className you can add in addition to `social-media-links`. */
+  className?: string;
+  /** Optional color */
+  color?: ColorType;
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Optional layout */
+  layout?: LayoutTypes;
+  /** Optional array of social media platform types, urls, and label texts */
+  linksData?: SocialMediaLinkDataProps[];
+  /** Optional true/false to display names of platform along with icon */
+  showLabels?: boolean;
+  /** Optional size: medium, large, xlarge */
+  size?: SizeType;
+}
+
+/* Accepts an array containing one or more data objects, each representing a social media platform and optional overrides for the label and/or URL.
+ *
+ *  @returns an array of data objects for each requested platform type that includes type, iconName, labelText and url where the latter two may include values supplied to override the defaults.
+ */
+function getLinksData(platforms: SocialMediaLinkDataProps[]) {
+  let allData = [];
+  platforms.forEach((myPlatform) => {
+    // Test if the type is allowed
+    if (!socialMediaLinkTypeArray.includes(myPlatform.type)) {
+      console.error(
+        "NYPL Reservoir SocialMediaLinks: " +
+          myPlatform.type +
+          " is either not a supported platform or is misspelled.\n" +
+          "Supported platforms include: " +
+          socialMediaLinkTypeArray.join(", ")
+      );
+      return;
+    }
+
+    // Get the dataset for this platform.
+    let thisPlatformArray = socialMediaDataMap.filter(
+      (socialMediaSlice) => socialMediaSlice.type === myPlatform.type
+    );
+
+    // The filter function returns an array of objects, so let's just get the objects out of the array.
+    let thisPlatformData = thisPlatformArray[0];
+
+    // If a url value exists in myPlatform use it, else use the default.
+    let newUrl = myPlatform.url ?? thisPlatformData.url;
+
+    // If a labelText value exists in myPlatform use it, else use the default.
+    let newLabelText = myPlatform.labelText ?? thisPlatformData.labelText;
+
+    let thisObj = {
```

Comment:
Can we rename this to be a bit more specific on its use? Maybe `modifiedPlatform`?